source/Halibut.Tests/MemoryFixture.cs

```diff
@@ -0,0 +1,181 @@
+﻿using System;
+using System.Diagnostics;
+using System.Net;
+using System.Net.Sockets;
+using System.Security.Cryptography.X509Certificates;
+using FluentAssertions;
+using Halibut.Diagnostics;
+using Halibut.ServiceModel;
+using JetBrains.dotMemoryUnit;
+using JetBrains.dotMemoryUnit.Kernel;
+using NUnit.Framework;
+
+namespace Halibut.Tests
+{
+    public interface ICalculatorService
+    {
+        long Add(long a, long b);
+    }
+
+    public class CalculatorService : ICalculatorService
+    {
+        public long Add(long a, long b)
+        {
+            //Console.WriteLine($"Got a request to add {a} + {b}");
+            return a + b;
+        }
+    }
+    
+    [TestFixture]
+    public class MemoryFixture
+    {
+        const int Clients = 5;
+        const int RequestsPerClient = 5;
+
+        [Test]
+        [DotMemoryUnit(SavingStrategy = SavingStrategy.OnCheckFail, Directory = @"c:\temp\dotmemoryunit", WorkspaceNumberLimit = 5, DiskSpaceLimit = 104857600)]
+        public void TcpClientsAreDisposedCorrectly()
+        {
+            if (!dotMemoryApi.IsEnabled)
+                Assert.Inconclusive("This test is meant to be run under dotMemory Unit. In your IDE, right click on the test icon and choose 'Run under dotMemory Unit'.");
+
+            using (RunServer(Certificates.Octopus, out var port))
+            {
+                for (var i = 0; i < Clients; i++)
+                    RunListeningClient(Certificates.TentacleListening, port, Certificates.OctopusPublicThumbprint);
+                for (var i = 0; i < Clients; i++)
+                    RunPollingClient(Certificates.TentaclePolling, Certificates.TentaclePollingPublicThumbprint);
+#if SUPPORTS_WEB_SOCKET_CLIENT
+                for (var i = 0; i < Clients; i++)
+                    RunWebSocketPollingClient(Certificates.TentaclePolling, Certificates.TentaclePollingPublicThumbprint, Certificates.OctopusPublicThumbprint);
+#endif
+
+                //https://dotnettools-support.jetbrains.com/hc/en-us/community/posts/360000088690-How-reproduce-DotMemory-s-Force-GC-button-s-behaviour-on-code-with-c-?page=1#community_comment_360000072750
+                for (var i = 0; i < 4; i++)
+                {
+                    GC.Collect(2, GCCollectionMode.Forced, true);
+                    GC.WaitForPendingFinalizers();
+                }
+
+                //server listen doesn't keep a port
```

Comment:
I have no idea why some of these _sometimes_ keep ports open, and sometimes dont.
Need an assist here.

Reply:
How long does it keep it open? There is some small amount of time Windows may keep it unavailable to prevent port hijack attacks.

Reply:
bad choice of words. I mean "an instance of `TcpClient` stays in memory"